ond.py

```diff
@@ -61,59 +63,85 @@ def run_protocol(self, config_):
-                        "FeatureExtraction", file_list
-                    )
+                algo_test_data["round_id"] = round_id
 
-                    results = {}
-                    results["detection"] = algorithm.execute(
-                        "WorldDetection", features_dict, logits_dict,
-                        red_light, round_id=round_id
-                    )
-                    results["classification"] = algorithm.execute(
-                        "NoveltyClassification", features_dict, logits_dict,
-                        round_id=round_id
+                try:
+                    dataset = self.interface.dataset_request(
+                        session_id, test_id, round_id
                     )
+                except RoundError:
+                    end_of_dataset = True
+                    continue
 
-                    if config["use_feedback"]:
-                        algorithm.execute("NoveltyAdaption", None)
-
-                    if config["save_features"]:
-                        logging.info(
-                            f"Writing features to {config['save_features']}"
-                        )
+                with open(dataset, "r") as dataset_:
+                    dataset_ids = dataset_.readlines()
+                    image_ids = [image_id.strip() for image_id in dataset_ids]
+                    algo_test_params["dataset_ids"].extend(image_ids)
 
-                    results["characterization"] = algorithm.execute(
-                        "NoveltyCharacterization", features_dict, logits_dict
-                    )
+                if config["use_saved_features"]:
+                    # TODO
+                    pass
                 else:
-                    end_of_dataset = True
+                    (
+                        algo_test_data["features_dict"],
+                        algo_test_data["logit_dict"]
+                    ) = algorithm.feature_extraction(algo_test_params)
+
+                    # TODO: save features
+
+                results = {}
+
+                results["detection"] = algorithm.world_detection(
+                    algo_test_params, algo_test_data
+                )
+
+                results["classification"] = algorithm.novelty_classification(
+                    algo_test_params, algo_test_data
+                )
+
+                # TODO: post results
+
+                if config["use_feedback"]:
+                    algorithm.novelty_adaption(algo_test_params, algo_test_data)
+
+                # TODO: round cleanup
 
                 round_id += 1
 
+            if config["save_features"]:
+                # TODO: save features
+                logging.info(
+                    f"Writing features to {config['save_features']}"
+                )
+
+            # TODO: save attributes
+
+            results["characterization"] = algorithm.novelty_characterization(
+                algo_test_params
```

Comment:
Shouldn't there be `algo_test_data` here too?

Reply:
Fixed in the latest commit.